Application/EdFi.Ods.Api/Middleware/XsdMetadataFileMiddleware.cs

```diff
@@ -47,21 +48,23 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
 
             var schema = routeValues["schema"].ToString();
 
-            var xsdFileInformationByUriSegment = _xsdFileInformationProvider.XsdFileInformationByUriSegment();
+            var xsdFileInformationByUriSegment = _xsdFileInformationProvider.XsdFileInformationByUriSegment(schema);
 
-            if (!xsdFileInformationByUriSegment.ContainsKey(schema))
+            if (xsdFileInformationByUriSegment == default)
             {
                 context.Response.StatusCode = StatusCodes.Status404NotFound;
                 return;
             }
 
-            string assemblyName = xsdFileInformationByUriSegment[schema].AssemblyName;
+            string assemblyName = xsdFileInformationByUriSegment.AssemblyName;
             string fullQualifiedFileName = $"{assemblyName}.Artifacts.Schemas.{routeValues["file"]}.xsd";
 
             context.Response.ContentType = "application/xml";
             context.Response.StatusCode = StatusCodes.Status200OK;
 
-            await context.Response.WriteAsync(_embeddedFileProvider.File(assemblyName, fullQualifiedFileName));
+            var streamReader = new StreamReader(_embeddedFileProvider.Stream(assemblyName, fullQualifiedFileName));
+
+            await context.Response.WriteAsync( await streamReader.ReadToEndAsync());
```

Comment:
If you're just going to read to end on the stream, you might as well not have a stream, you're just wrapping more objects around the process of reading the entire file to memory.  If we want to do streams we need to return a stream response.  I think what we're looking for is something like https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.sendfileresponseextensions.sendfileasync?view=aspnetcore-3.1?

Reply:
done